go.mod

```diff
@@ -0,0 +1,130 @@
+	github.com/coreos/go-iptables v0.4.5
+	github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf // indirect
+	github.com/davecgh/go-spew v1.1.1
+	github.com/go-martini/martini v0.0.0-20170121215854-22fa46961aab // indirect
+	github.com/go-sql-driver/mysql v1.5.0
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
+	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e // indirect
+	github.com/golang/protobuf v1.3.4
+	github.com/google/uuid v1.1.1
+	github.com/gorilla/mux v1.7.0
+	github.com/gorilla/websocket v1.4.1
+	github.com/grpc-ecosystem/go-grpc-middleware v1.2.0 // indirect
+	github.com/grpc-ecosystem/grpc-gateway v1.14.1 // indirect
+	github.com/hashicorp/consul/api v1.4.0 // indirect
+	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
+	github.com/martini-contrib/render v0.0.0-20150707142108-ec18f8345a11 // indirect
+	github.com/mattn/go-sqlite3 v2.0.3+incompatible
+	github.com/natefinch/pie v0.0.0-20170715172608-9a0d72014007
+	github.com/nu7hatch/gouuid v0.0.0-20131221200532-179d4d0c4d8d
+	github.com/openshift/api v0.0.0-20200217161739-c99157bc6492
+	github.com/oxtoacart/bpool v0.0.0-20190530202638-03653db5a59c // indirect
+	github.com/pkg/errors v0.8.1
+	github.com/prometheus/client_golang v1.5.0
+	github.com/prometheus/procfs v0.0.10 // indirect
+	github.com/sirupsen/logrus v1.4.2
+	github.com/socketplane/libovsdb v0.0.0-20170116174820-4de3618546de
+	github.com/spf13/cobra v0.0.6
+	github.com/spf13/viper v1.4.0
+	github.com/stretchr/testify v1.4.0
+	github.com/tatsushid/go-fastping v0.0.0-20160109021039-d7bb493dee3e
+	github.com/tedsuo/ifrit v0.0.0-20191009134036-9a97d0632f00 // indirect
+	github.com/tedsuo/rata v1.0.0 // indirect
+	github.com/tmc/grpc-websocket-proxy v0.0.0-20200122045848-3419fae592fc // indirect
+	github.com/ugorji/go v1.1.7 // indirect
+	github.com/vishvananda/netlink v1.0.0
+	github.com/vito/go-sse v1.0.0 // indirect
+	github.com/yl2chen/cidranger v1.0.0
+	go.etcd.io/etcd v0.5.0-alpha.5.0.20190911215424-9ed5f76dc03b
+	go.uber.org/multierr v1.5.0 // indirect
+	go.uber.org/zap v1.14.0 // indirect
+	golang.org/x/crypto v0.0.0-20200302210943-78000ba7a073 // indirect
+	golang.org/x/lint v0.0.0-20200302205851-738671d3881b // indirect
+	golang.org/x/net v0.0.0-20200301022130-244492dfa37a
+	golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527 // indirect
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0
+	golang.org/x/tools v0.0.0-20200306191617-51e69f71924f // indirect
+	google.golang.org/genproto v0.0.0-20200306153348-d950eab6f860 // indirect
+	google.golang.org/grpc v1.27.1
+	k8s.io/api v0.17.3
+	k8s.io/apimachinery v0.17.3
+	k8s.io/client-go v8.0.0+incompatible
+	k8s.io/kubernetes v0.0.0-00010101000000-000000000000
+	sigs.k8s.io/controller-runtime v0.5.0
+)
+
+replace (
+	go.etcd.io/bbolt => go.etcd.io/bbolt v1.3.3
+	k8s.io/api => k8s.io/kubernetes/staging/src/k8s.io/api v0.0.0-20200211180626-06ad960bfd03
```

Comment:
After doing this, and some of the other version pining below, to what extent will we be compatible with an older version of kubernetes API/server? There will be users who will have deployed k8s < 1.17, either directly, or by virtue of using a distro like OpenShift/RH, and we wont be able to force them to upgrade their k8s deployment.

Reply:
Sumit, it should be backward compatible, may be we should validate, if we have any 1.13 older version fabs

Reply:
BTW Currently we are using 1.15 API version

Reply:
Why just 1.13? Elsewhere below you have version pinned to 1.17. So what about any version < 1.17? We will have fewer, perhaps if any customers on, say 1.13, but we will definitely customers on 1.16 (also need to confirm what OpenShift 3.11 has).